ocaml/xapi/xapi_vm_snapshot.ml

```diff
@@ -203,12 +203,13 @@ let checkpoint ~__context ~vm ~new_name =
 		end;
 
 		(* snapshot the disks and the suspend VDI *)
-		let snap =
+		let snap, err =
 			if not (TaskHelper.is_cancelling ~__context) then begin
-				try Some (Xapi_vm_clone.clone Xapi_vm_clone.Disk_op_checkpoint ~__context ~vm ~new_name ~snapshot_info_record:!snapshot_info)
```

Comment:
Ah ha! I see now. The call to `Xapi_xenops.resume` was not in the context of the patch. Only visible when expanding the file.